### ISSUE TITLE
For the manipulation commands, conform to users’ style preferences

### ISSUE THREAD

**Issue by jotaen** [jotaen/klog#129]
# Problem
When using the `track`, `start`, … subcommands, klog detects the indentation style used in the file and takes that over for the newly created entry automatically.

It would be nice to also take other style preferences into account:

- How the date is formatted (`2020/01/01` or `2020-01-01`)
- Whether 12-hour clock times are used (`1:00pm` instead of `13:00`)
- Whether or not there are spaces around the dash (`1:00-?` vs. `1:00 - ?`)

Instead, klog currently just defaults to the style that is recommended via the spec. (`2020-01-01`, `13:00`, `1:00 - ?`.)

# Potential solutions

I’ll probably start with on-the-fly, because that’s nice in itself and independent of an explicit config. A proper config can still be added later, though I have to say I’m pretty reluctant towards it.

## On-the-fly detection
Basically the same as the indentation detection.
- Pro: lightweight and easy-to-use (since it “just works” without further ado)
- Con: doesn’t have an effect on new (empty) files

## Environment variables
Like `KLOG_USE_AM_PM` etc.
- Pro: works for empty files; is explicit
- Con: only works within the realm of the own bash profile

## Config file
Something like `~/.klog/config.yaml` where the user can (optionally) set their preferences. Instead of using YAML/TOML/etc., it could also just be `~/.klog/sample.klg`.
- Pro: works for empty files; is most explicit
- Con: feels more heavyweight (needs to be documented properly; how to behave when config is malformed/invalid; etc.)
